File: 0905_sort_array_by_parity.swift
// https://leetcode.com/problems/sort-array-by-parity/

class Solution {
    func sortArrayByParity(_ nums: [Int]) -> [Int] {
        var evens = [Int]()
        var odds = [Int]()

        for num in nums {
            if num % 2 == 0 {
                evens.append(num)
            } else {
                odds.append(num)
            }
        }

        return evens + odds
    }
}

File: 0905_sort_array_by_parity.kt
// https://leetcode.com/problems/sort-array-by-parity/

class Solution {
    fun sortArrayByParity(nums: IntArray): IntArray {
        val numsList = nums.toMutableList()
        val n = numsList.size
        var i = 0
        for (j in 0 until n) {
            val num = numsList[i]
            if (num % 2 == 0) {
                i++
            } else {
                numsList.removeAt(i)
                numsList.add(num)
            }
        }
        return numsList.toIntArray()
    }
}
